driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -464,6 +480,9 @@ else if (chosen.state.compareAndSet(TRASHED, RESURRECTING))
     }
 
     private void maybeSpawnNewConnection() {
+        if (!host.convictionPolicy.canReconnectNow())
```

Comment:
What if we have been naturally growing/shrinking connections?  Spawning seems like a good idea if connections aren't being closed for bad reasons.